lib/commands/file-movement.js

```diff
@@ -93,6 +71,49 @@ async function createService (udid, remotePath) {
   }
 }
 
+async function pullFileFromRealDevice (service, relativePath) {
+  const stream = await service.createReadStream(relativePath, { autoDestroy: true });
+  const closeEvent = new B((resolve) => stream.on('close', resolve)).timeout(IO_TIMEOUT);
+  const buffer = [];
+  stream.on('data', (data) => buffer.push(data));
+  try {
+    await closeEvent;
+  } catch (e) {
+    throw new Error(`Couldn't pull the file within the given timeout ${IO_TIMEOUT}ms`);
+  }
+  return Buffer.concat(buffer).toString('base64');
+}
+
+
+async function pullFolderFromRealDevice (service, relativePath) {
+  const tmpFolder = await tempDir.openDir();
+  try {
+    const folderPath = path.join(tmpFolder, relativePath);
+    await mkdirp(folderPath);
+    const promises = [];
+    await service.walkDir(relativePath, true, async (itemPath, isDir) => {
+      const pathOnServer = path.join(tmpFolder, itemPath);
```

Comment:
path.posix.join

Explanation:

The `path` module is platform dependent. So for example, if the code is running on Windows, the path separater is always a backslash, which is not acceptable on iOS/Android, where a normal slash is used. That is why it is always safe to explicitly set `path.posix`, so the target path is constructed properly independently of the host OS

Reply:
@mykola-mokhnach I'm aware of that. I left it path.join since the appium server can be run also on windows. I can change it to posix if we want to

Reply:
@mykola-mokhnach Should I do it? I'm confused if I should or not do it atm

Reply:
This should be done if `path` module is used to construct/handle paths on the target phone. If we are handling paths on the local file system where the Appium server is running then everything is fine